radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/messaging/InboundMessage.java

```diff
@@ -65,72 +65,31 @@
 package com.radixdlt.network.messaging;
 
 import com.radixdlt.network.p2p.NodeId;
-import com.radixdlt.utils.Bytes;
 
 import java.util.Arrays;
 import java.util.Objects;
 
-/**
- * A raw message received from a peer, before decoding.
- */
-public final class InboundMessage {
-	private final NodeId source;
-	private final byte[] message;
-
-	/**
-	 * Creates an inbound message with the specified source and message.
-	 *
-	 * @param source The source of the message.
-	 * @param message The message received.
-	 * @return a constructed {@code InboundMessage}
-	 */
-	public static InboundMessage of(NodeId source, byte[] message) {
-		return new InboundMessage(source, message);
-	}
-
-	private InboundMessage(NodeId source, byte[] message) {
-		// Null checking not performed for high-frequency interface
-		this.source = source;
-		this.message = message;
-	}
-
-	/**
-	 * Returns the source of the message.
-	 *
-	 * @return the source of the message.
-	 */
-	public NodeId source() {
-		return source;
-	}
-
-	/**
-	 * Returns the message.
-	 *
-	 * @return the message.
-	 */
-	public byte[] message() {
-		return message;
-	}
-
+public record InboundMessage(long receiveTime, NodeId source, byte[] message) {
 	@Override
 	public int hashCode() {
-		return Arrays.hashCode(message) * 31 + Objects.hashCode(source);
+		return Objects.hash(receiveTime, source, Arrays.hashCode(message));
 	}
 
 	@Override
 	public boolean equals(Object obj) {
 		if (this == obj) {
 			return true;
 		}
-		if (obj instanceof InboundMessage) {
-			final var other = (InboundMessage) obj;
-			return Objects.equals(this.source, other.source) && Arrays.equals(this.message, other.message);
+		if (obj instanceof InboundMessage other) {
+			return receiveTime == other.receiveTime
+				&& Objects.equals(this.source, other.source)
+				&& Arrays.equals(this.message, other.message);
 		}
 		return false;
 	}
 
 	@Override
 	public String toString() {
-		return String.format("%s[%s:%s]", getClass().getSimpleName(), source, Bytes.toHexString(message));
+		return String.format("%s[receiveTime=%s,source=%s]", getClass().getSimpleName(), receiveTime, source);
 	}
 }
```

Comment:
These methods are generated by the compiler

Reply:
yeah but the generated implementation is a bit more noisy (it includes the meaningless hashcode of message bytes)

Reply:
But you have the same in your hashCode implementation. Also, I really see no point to switching to records without using default hashCode and equals. After all, it was one of the motivations behind the introduction of the records.

Reply:
this needs some more investigation, I don't know why but EqualsVerifier fails for the record's generated equals/hashCode

Reply:
but even with these being override, using records saves some boilerplate on getters

Reply:
I believe that with records, we can remove test with `EqualsVerifier`. The intent of this test is to ensure that `equals()` and `hashCode()` are written correctly, but this is not necessary when the compiler is responsible for this part. So, my suggestion is to remove both, test and implementations for these methods.

Reply:
ok, done